File: templates/ProjectTemplate/VisualBasic/1033/MobileServicesApplicationv1.0/App_Start/Startup.MobileApp.vb
Imports System
Imports System.Collections.Generic
Imports System.Data.Entity
Imports System.Web.Http
Imports Microsoft.Azure.Mobile.Server.Config
Imports Owin

Partial Public Class Startup
    Public Sub ConfigureMobileApp(ByVal app As IAppBuilder)

        Dim config As New HttpConfiguration()

        Dim mobileConfig As New MobileAppConfiguration()
        mobileConfig _
            .UseDefaultConfiguration() _
            .ApplyTo(config)

        Database.SetInitializer(New $safeinitializerclassname$())

        app.UseMobileAppAuthentication(config)
        app.UseWebApi(config)

    End Sub
End Class

Public Class $safeinitializerclassname$
    Inherits CreateDatabaseIfNotExists(Of $safecontextclassname$)

    Protected Overrides Sub Seed(ByVal context As $safecontextclassname$)
        Dim todoItems As List(Of TodoItem) = New List(Of TodoItem) From
            {
                New TodoItem With {.Id = Guid.NewGuid().ToString(), .Text = "First item", .Complete = False},
                New TodoItem With {.Id = Guid.NewGuid().ToString(), .Text = "Second item", .Complete = False}
            }

        For Each todoItem As TodoItem In todoItems
            context.Set(Of TodoItem).Add(todoItem)
        Next

        MyBase.Seed(context)

    End Sub
End Class

File: templates/ProjectTemplate/CSharp/1033/MobileServicesApplicationv1.0/App_Start/Startup.MobileApp.cs
using System;
using System.Collections.Generic;
using System.Data.Entity;
using System.Web.Http;
using Microsoft.Azure.Mobile.Server.Config;
using Owin;
using $safeprojectname$.DataObjects;
using $safeprojectname$.Models;

namespace $safeprojectname$
{
    public partial class Startup
    {
        public static void ConfigureMobileApp(IAppBuilder app)
        {
            HttpConfiguration config = new HttpConfiguration();

            new MobileAppConfiguration()
                .UseDefaultConfiguration()
                .ApplyTo(config);
            
            Database.SetInitializer(new $safeinitializerclassname$());

            app.UseMobileAppAuthentication(config);
            app.UseWebApi(config);
        }
    }

    public class $safeinitializerclassname$ : CreateDatabaseIfNotExists<$safecontextclassname$>
    {
        protected override void Seed($safecontextclassname$ context)
        {
            List<TodoItem> todoItems = new List<TodoItem>
            {
                new TodoItem { Id = Guid.NewGuid().ToString(), Text = "First item", Complete = false },
                new TodoItem { Id = Guid.NewGuid().ToString(), Text = "Second item", Complete = false }
            };

            foreach (TodoItem todoItem in todoItems)
            {
                context.Set<TodoItem>().Add(todoItem);
            }

            base.Seed(context);
        }
    }
}
